File: nosh_now_apis/Models/Order.cs
using Microsoft.VisualStudio.TextTemplating;

namespace MyApp.Models
{
    public class Order : IModel
    {
        public int Id { get; set; }
        public DateTime OrderedDate { get; set; }
        public double ShipmentFee { get; set; }
        public string? Phone { get; set; }
        public string? Coordinator { get; set; }
        public int StatusId { get; set; }
        public int MerchantId { get; set; }
        public int EaterId { get; set; }
        public int? ShipperId { get; set; }
        public int? MethodId { get; set; }
        public virtual Eater Eater { get; set; }
        public virtual OrderStatus Status { get; set; }
        public virtual Merchant Merchant { get; set; }
        public virtual Shipper Shipper { get; set; }
        public virtual PaymentMethod PaymentMethod { get; set; }
        public virtual ICollection<OrderDetail> OrderDetails { get; set; }


        public double CalcTotal()
        {
            double total = 0;
            foreach (OrderDetail detail in OrderDetails)
            {
                total += detail.Price * detail.Quantity;
            }
            return total + ShipmentFee;
        }
        new public String ToString()
        {
            return $"Id: {Id}, OrderedDate: {OrderedDate}, ShipmentFee: {ShipmentFee},totalPay: {0} Phone: {Phone}, Coordinator: {Coordinator}, StatusId: {StatusId}, MerchantId: {MerchantId}, EaterId: {EaterId}, ShipperId: {ShipperId}, MethodID: {MethodId}";
        }
    }
}

File: nosh_now_application/lib/data/models/order.dart
import 'package:nosh_now_application/data/models/eater.dart';
import 'package:nosh_now_application/data/models/merchant.dart';
import 'package:nosh_now_application/data/models/order_status.dart';
import 'package:nosh_now_application/data/models/payment_method.dart';
import 'package:nosh_now_application/data/models/shipper.dart';

class Order {
  int orderId;
  DateTime? orderedDate;
  double? shipmentFee;
  double totalPay;
  String? coordinator;
  String? phone;
  OrderStatus orderStatus;
  Merchant merchant;
  Eater eater;
  Shipper? shipper;
  PaymentMethod? paymentMethod;
  Order(
      {required this.orderId,
      required this.totalPay,
      required this.orderStatus,
      required this.eater,
      required this.merchant,
      this.coordinator,
      this.orderedDate,
      this.paymentMethod,
      this.phone,
      this.shipmentFee,
      this.shipper});
  factory Order.fromJson(Map<dynamic, dynamic> json) {
    return Order(
        orderId: json['id'],
        totalPay: json['totalPay'] / 1.0,
        orderStatus: OrderStatus.fromJson(json['status']),
        eater: Eater.fromJson(json['eater']),
        merchant: Merchant.fromJson(json['merchant']),
        coordinator: json['coordinator'] ?? json['coordinator'],
        phone: json['phone'] ?? json['phone'],
        orderedDate: DateTime.parse(json['orderedDate']),
        shipmentFee: json['shipmentFee'] / 1.0,
        paymentMethod: json['paymentMethod'] != null
            ? PaymentMethod.fromJson(json['paymentMethod'])
            : null,
        shipper:
            json['shipper'] != null ? Shipper.fromJson(json['shipper']) : null);
  }
}
